test/units/module_utils/facts/system/test_date_time.py

```diff
@@ -0,0 +1,79 @@
+# unit tests for ansible system date_time fact collectors
+# -*- coding: utf-8 -*-
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import datetime
+import time
+
+from units.compat.mock import Mock, patch
+from .. base import BaseFactsTest
+
+from ansible.module_utils.facts.system.date_time import DateTimeFactCollector
+
+
+class MockDatetime(datetime.datetime):
+    @classmethod
+    def now(cls):
+        return cls(2020, 7, 8, 12, 34, 56, 123456)
+
+    @classmethod
+    def utcnow(cls):
+        return cls(2020, 7, 8, 16, 34, 56, 123456)
+
+
+class TestDateTimeFacts(BaseFactsTest):
+    __test__ = True
+    gather_subset = ['!all', 'date_time']
+    valid_subsets = ['date_time']
+    fact_namespace = 'ansible_date_time'
+    collector_class = DateTimeFactCollector
+
+    def _mock_module(self):
+        mock_module = Mock()
+        mock_module.params = {'gather_subset': self.gather_subset,
+                              'gather_timeout': 10,
+                              'filter': '*'}
+        return mock_module
+
+    def test_collect(self):
+        module = self._mock_module()
+        datetime.datetime = MockDatetime
+        fact_collector = self.collector_class()
+        facts_dict = fact_collector.collect(module=module)
+
+        self.assertIsInstance(facts_dict, dict)
+        self.assertEqual(facts_dict['date_time']['year'], '2020')
+        self.assertEqual(facts_dict['date_time']['month'], '07')
+        self.assertEqual(facts_dict['date_time']['weekday'], 'Wednesday')
```

Comment:
Under the pytest env it's recommended to utilize the `assert` statement rather than relying on unittest helper methods, for example:
```suggestion
        assert isinstance(facts_dict, dict)
        assert facts_dict['date_time']['year'] == '2020'
        assert facts_dict['date_time']['month'] == '07'
        assert facts_dict['date_time']['weekday'] == 'Wednesday'
```